apps/web/src/app/(authenticated)/insights/components/order-filters.tsx

```diff
@@ -109,6 +121,29 @@ export default function OrderFilters(): React.ReactNode {
     });
   };
 
+  const handleDateRangeChange = (dates: DatesRangeValue): void => {
+    const newParams = new URLSearchParams(searchParams.toString());
+    const dateFrom = dates[0]?.getTime();
+    const dateTo = dates[1]?.getTime();
+    if (dateFrom && dateTo) {
+      resetInsights();
+      newParams.set(dateFromKey, String(dateFrom));
+      newParams.set(dateToKey, String(dateTo));
+      const newURL = `${pathname}?${newParams.toString()}`;
+      startTransition(() => {
+        router.replace(newURL);
```

Comment:
```suggestion
        window.history.replaceState(null, '', newURL)
```
Here are the docs: https://nextjs.org/docs/app/building-your-application/routing/linking-and-navigating#using-the-native-history-api you can replace the url without reloading the page.

Which means that if we have the insights call in a useEffect that is dependent only on searchparams, we should be re-rendering only the changed component and the insights area.

Reply:
HUGE

Reply:
Actually, I tried it on the page size dropdown and seems to be rerendering other stuff too :/
Also, doesn't seem to re-fetch data and shows "no results".

Reply:
Yes I know there are some issues, but I am assuming it is this neighbouring issue that triggers the re-render. I don't know what would be causing the no re-fetch though. We probably need to test this on a clean project